### PR TITLE
Add Logging trace_sampled

### LINKED ISSUE #2690: Support the LogEntry "trace_sampled" field in GoogleCloudPlatform/fluent-plugin-google-cloud
https://github.com/googleapis/googleapis/commit/7b34e5a3ed10775fe14d15ffa842870dc3b5400e added the `trace_sampled` field to the Stackdriver `LogEntry` proto.  It would be nice if the Stackdriver Logging agent handled `trace_sampled` similarly to the `trace` and `span_id` fields and used a special field in the Fluentd log event to set `LogEntry.trace_sampled` (See https://cloud.google.com/logging/docs/agent/configuration#special_fields_in_structured_payloads).  This feature could be used in combination with OpenCensus log correlation to allow Stackdriver to interpret the sampling decision from OpenCensus.

This change would be very similar to https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/180.

### PR DESCRIPTION
**Pull request by blowmage**
* Add `Entry#trace_sampled` attribute
* Add `Logger::RequestInfo#trace_sampled` attribute

[fixes #2690]

### CODE REVIEW

File: google-cloud-logging/test/helper.rb [hunk 1]
```diff
@@ -73,14 +73,15 @@ def random_entry_hash
       "severity"  => :DEFAULT,
       "insert_id" => "abc123",
       "labels" => {
-        "env" => "production",
-        "foo" => "bar"
+        "env"  => "production",
+        "foo"  => "bar"
```
Comment (dazuma) [inline google-cloud-logging/test/helper.rb]: Nit: Is this necessary?

Comment (blowmage) [reply google-cloud-logging/test/helper.rb]: No. None of the hash spacing is neccessary. But I think it looks better when it is consistent across lines.

Comment (dazuma) [reply google-cloud-logging/test/helper.rb]: Usually I see the spacing being different at different "levels" of the hash. But, meh, LGTM or whatever. :)

**Comment by qingling128** [comment]
Hi @blowmage, any chance we have a `google-cloud-logging` release coming that might include this change?

**Pull request merged**